ql/src/test/org/apache/hadoop/hive/ql/exec/TestGetPartitionAuthWithBatches.java

```diff
@@ -0,0 +1,274 @@
+import org.apache.hadoop.hive.ql.metadata.HiveException;
+import org.apache.hadoop.hive.ql.metadata.Partition;
+import org.apache.hadoop.hive.ql.session.SessionState;
+import org.junit.Assert;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+
+import java.util.Arrays;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+public class TestGetPartitionAuthWithBatches {
+
+    private final String catName = "hive";
+    private final String dbName = "default";
+    private final String tableName = "test_partition_batch_with_auth";
+    private static HiveConf hiveConf;
+    private static HiveMetaStoreClient msc;
+    private static Hive hive;
+    private Table table;
+
+    @BeforeClass
+    public static void setupClass() throws HiveException {
+        hiveConf = new HiveConf(TestGetPartitionAuthWithBatches.class);
+        hiveConf.set("hive.security.authorization.enabled", "true");
+        hiveConf.set("hive.security.authorization.manager","org.apache.hadoop.hive.ql.security.authorization.DefaultHiveAuthorizationProvider");
+        hive = Hive.get();
+        SessionState.start(hiveConf);
+        try {
+            msc = new HiveMetaStoreClient(hiveConf);
+        } catch (MetaException e) {
+            throw new HiveException(e);
+        }
+    }
+
+    @Before
+    public void before() throws Exception {
+        PartitionUtil.createPartitionedTable(msc, catName, dbName, tableName);
+        table = msc.getTable(catName, dbName, tableName);
+        PartitionUtil.addPartitions(msc, dbName, tableName, table.getSd().getLocation(), hiveConf);
+    }
+
+    @After
+    public void after() throws Exception {
+        PartitionUtil.cleanUpTableQuietly(msc, catName, dbName, tableName);
+    }
+
+    @Test
+    public void testNumberOfPartitionsRetrieved() throws HiveException {
+        List<String> numParts = hive.getPartitionNames(dbName, tableName, (short)-1);
+        Assert.assertEquals(numParts.size(), 30);
+        List<Partition> partitions = hive.getPartitionsAuthByNames(new org.apache.hadoop.hive.ql.metadata.Table(table),
+                numParts.subList(0,5), "username", new ArrayList<>(Arrays.asList("Grp1", "Grp2")));
+        Assert.assertEquals(partitions.size(), 5);
+    }
+
+    /**
+     * Tests the number of partitions recieved from the HMS
+     *
+     * @throws Exception
+     */
+    @Test
+    public void testGetPartitionsAPI() throws Exception {
+        List<org.apache.hadoop.hive.ql.metadata.Partition> part = hive.getPartitions(hive.getTable(dbName, tableName));
+        Assert.assertEquals(part.size(), 30);
```

Comment:
You may want to define const for partition count

Reply:
done